src/op_handlers/fat_pointer_read.rs

```diff
@@ -10,6 +10,13 @@ pub fn fat_pointer_read(vm: &mut VMState, opcode: &Opcode) -> Result<(), EraVmEr
     if !src0.is_pointer {
         return Err(OperandError::InvalidSrcNotPointer(opcode.variant).into());
     }
+
+    // make sure the pointer is a value that can be added without overflow for 32 bits.
+    if src0.value.low_u32() > (u32::MAX - 32) {
+        vm.set_gas_left(0)?;
+        return Err(HeapError::InvalidAddress.into());
```

Comment:
Maybe we should `panic_from_far_call` here? Does that helper take the error code to forward it?

Reply:
Yes, any err returned from far_call will trigger the`panic_from_far_call`. See [here](https://github.com/lambdaclass/era_vm/blob/main/src/lib.rs#L179C17-L185C24).